blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/RollupService.java

```diff
@@ -204,8 +204,14 @@ final void poll() {
     }
 
     public void run() {
+        run(null, null);
+    }
+    public void run(Long maxNumIterations, Long maxMillisRunTime) {
```

Comment:
My principal is not to change too much of the main code/functionality for just tests. I understand sometimes moving code around (like breaking a big method to smaller methods) so we can test properly is sometimes necessary. I am ok with that example of breaking a big method to smaller ones. But at the very least, we should avoid changing the functionality.

This change introduces new functionality of the main code, for the purpose of tests. I am having troubles justifying it.

A quick google found me this. Now... I admit, I haven't tried it. I don't know for sure if it would work. But this should hopefully give you maybe some ideas:

http://stackoverflow.com/questions/17206527/is-it-possible-to-verify-a-mock-method-running-in-different-thread-in-mockito

Should look at JMockit too, to see if it has something like this. I'd rather us use just one mocking tool, but if it's not the end of the world if we use JMockit and Mockito both.